src/middlewared/middlewared/plugins/network.py

```diff
@@ -2350,7 +2351,26 @@ async def sync(self):
             # Otherwise change it
             if not routing_table.default_route_ipv4:
                 self.logger.info('Adding IPv4 default route to {}'.format(ipv4_gateway.gateway))
-                routing_table.add(ipv4_gateway)
+                try:
+                    routing_table.add(ipv4_gateway)
+                except NetlinkError as e:
+                    if 101 in e.args:  # Network host unreachable
+                        # This error occurs in random race conditions.
+                        # For example, can occur in the following scenario:
+                        #   1. delete all configured interfaces on system
+                        #   2. interface.sync() gets called and starts dhcp
+                        #       on all interfaces detected on the system
+                        #   3. route.sync() gets called which eventually
+                        #       calls dhclient_leases which reads a file on
+                        #       disk to see if we have any previously
+                        #       defined default gateways from dhclient.
+                        #       However, by the time we read this file,
+                        #       dhclient could still be requesting an
+                        #       address from the DHCP server
+                        #   4. so when we try to install our own default
+                        #       gateway manually (even though dhclient will
+                        #       do this for us) it will fail expectedly here.
+                        pass
```

Comment:
Let's please log this at least for scenarios where the race condition does not hold and we have a gateway configured already which cannot be added in the current system network state.